src/MarginTrading.AssetService.Services/ProductsService.cs

```diff
@@ -143,19 +143,23 @@ public async Task<Result<ProductsErrorCodes>> DeleteAsync(string productId, stri
         public Task<Result<Product, ProductsErrorCodes>> GetByIdAsync(string productId)
             => _repository.GetByIdAsync(productId);
 
-        public Task<Result<List<Product>, ProductsErrorCodes>> GetAllAsync(string[] mdsCodes, string[] productIds,
+        public Task<Result<List<Product>, ProductsErrorCodes>> GetAllAsync(
+            string mdsCodeFilter,
+            string[] mdsCodes, string[] productIds,
             bool? isStarted, bool? isDiscontinued)
-            => _repository.GetAllAsync(mdsCodes, productIds, isStarted, isDiscontinued);
+            => _repository.GetAllAsync(mdsCodeFilter, mdsCodes, productIds, isStarted, isDiscontinued);
 
-        public Task<Result<List<Product>, ProductsErrorCodes>> GetByPageAsync(string[] mdsCodes, string[] productIds,
+        public Task<Result<List<Product>, ProductsErrorCodes>> GetByPageAsync(
+            string mdsCodeFilter,
+            string[] mdsCodes, string[] productIds,
             bool? isStarted,
             bool? isDiscontinued,
             int skip = default, int take = 20)
-            => _repository.GetByPageAsync(mdsCodes, productIds, isStarted, isDiscontinued, skip, take);
+            => _repository.GetByPageAsync(mdsCodeFilter, mdsCodes, productIds, isStarted, isDiscontinued, skip, take);
 
         public async Task<Result<ProductsErrorCodes>> UpdateBatchAsync(List<Product> products, string username)
         {
-            var existing = await _repository.GetAllAsync(null, products.Select(p => p.ProductId).ToArray());
+            var existing = await _repository.GetAllAsync(null, null, products.Select(p => p.ProductId).ToArray());
```

Comment:
It would make sense to use the named argument pattern here for better readability.

```suggestion
            var existing = await _repository.GetAllAsync(mdsCodeFilter: null, mdsCodes: null, products.Select(p => p.ProductId).ToArray());
```